app/src/main/java/io/appium/uiautomator2/utils/ElementHelpers.java

```diff
@@ -80,4 +87,56 @@ private static AccessibilityNodeInfo elementToNode(Object element) {
     public static JSONObject toJSON(AndroidElement el) throws JSONException {
         return new JSONObject().put("ELEMENT", el.getId());
     }
+
+    /**
+     * Set text of an element
+     *
+     * @param element - target element
+     * @param text - desired text
+     * @param unicodeKeyboard - true, if text should be encoded to unicode
+     */
+    public static void setText(final Object element, final String text, final boolean unicodeKeyboard) throws UiObjectNotFoundException {
+        String textToSend = text;
+        AccessibilityNodeInfo nodeInfo = AccessibilityNodeInfoGetter.fromUiObject(element);
+
+        /**
+         * Execute ACTION_SET_PROGRESS action (introduced in API level 24)
+         * if element has range info and text can be converted to float.
+         * Falling back to element.setText() if something goes wrong.
+         */
+        if (nodeInfo.getRangeInfo() != null && Build.VERSION.SDK_INT >= 24) {
+            Logger.debug("Element has range info.");
+            try {
+                if (AccessibilityNodeInfoHelper.setProgressValue(nodeInfo, Float.parseFloat(text))) {
+                    return;
+                }
+            } catch (NumberFormatException e) {
+                Logger.debug(String.format("Can not convert \"%s\" to float.", text));
```

Comment:
intercept and re-throw it to as UiAutomator2Exception instead of writing the log